Add support for cluster validations in gRPC clients (#10788)
grafana/mimir: CHANGELOG.md

  * `-server.cluster-validation.grpc.enabled`
  * `-server.cluster-validation.grpc.soft-validation`
* [ENHANCEMENT] All: Add `cortex_client_request_invalid_cluster_validation_labels_total` metrics, that is used by Mimir's gRPC clients to track invalid cluster validations. #10767

> * [ENHANCEMENT] gRPC clients: add support to configure the cluster validation label to be sent together with the requests, via an experimental common configuration `-common.client-cluster-validation.label`. When set, this configuration will be inherited by the following gRPC clients: #10788

Reviewer: ```suggestion
* [ENHANCEMENT] gRPC clients: add support to configure the cluster validation label to be sent together with the requests, via an experimental common configuration `-common.client-cluster-validation.label`. When set, this configuration is inherited by the following gRPC clients: #10788
```

>   * ingester clients, used for communications between the distributors, queriers or rulers and the ingesters,

Reviewer: ```suggestion
  * ingester clients, used for communication between the distributors, queriers, or rulers and the ingesters
```

Reviewer: We don't need to separate list items with commas

>   * store-gateway clients, used for communications between the queriers and the store-gateways,

Reviewer: ```suggestion
  * store-gateway clients, used for communication between the queriers and the store-gateways
```

>   * query-frontened clients, used for communications between the queriers and the query-frontends,

Reviewer: ```suggestion
  * query-frontened clients, used for communication between the queriers and the query-frontends
```

>   * query-frontened clients, used for communications between the query-schedulers and the query-frontends,

Reviewer: ```suggestion
  * query-frontened clients, used for communication between the query-schedulers and the query-frontends
```

>   * query-frontened clients, used for communications between the rulers and the query-frontends,

Reviewer: ```suggestion
  * query-frontend clients, used for communication between the rulers and the query-frontends
```

>   * query-scheduler clients, used for communications between the queriers and the query-schedulers,

Reviewer: ```suggestion
  * query-scheduler clients, used for communication between the queriers and the query-schedulers
```

>   * query-scheduler clients, used for communications between the frontend-queriers and the query-schedulers,

Reviewer: ```suggestion
  * query-scheduler clients, used for communication between the frontend-queriers and the query-schedulers
```

>   * ruler clients, used for communications between different ruler instances,

Reviewer: ```suggestion
  * ruler clients, used for communication between different ruler instances
```

>   * block-builder-scheduler clients, used for communications between the block-builders and the block-builder-schedulers.

Reviewer: ```suggestion
  * block-builder-scheduler clients, used for communication between the block-builders and the block-builder-schedulers
```
